test/core/load_balancing/bm_picker.cc

```diff
@@ -0,0 +1,260 @@
+      work_serializer_->Schedule(
+          [this, num_endpoints]() {
+            EndpointAddressesList addresses;
+            for (size_t i = 0; i < num_endpoints; i++) {
+              grpc_resolved_address addr;
+              int port = i % 65536;
+              int ip = i / 65536;
+              CHECK(ip < 256);
+              CHECK(grpc_parse_uri(
+                  URI::Parse(absl::StrCat("ipv4:127.0.0.", ip, ":", port))
+                      .value(),
+                  &addr));
+              addresses.emplace_back(addr, ChannelArgs());
+            }
+            CHECK_OK(lb_policy_->UpdateLocked(LoadBalancingPolicy::UpdateArgs{
+                std::make_shared<EndpointAddressesListIterator>(
+                    std::move(addresses)),
+                config_, "", ChannelArgs()}));
+          },
+          DEBUG_LOCATION);
+    }
+    work_serializer_->DrainQueue();
+  }
+
+ private:
+  class SubchannelFake final : public SubchannelInterface {
+   public:
+    explicit SubchannelFake(BenchmarkHelper* helper) : helper_(helper) {}
+
+    void WatchConnectivityState(
+        std::unique_ptr<ConnectivityStateWatcherInterface> unique_watcher)
+        override {
+      AddConnectivityWatcherInternal(
+          std::shared_ptr<ConnectivityStateWatcherInterface>(
+              std::move(unique_watcher)));
+    }
+
+    void CancelConnectivityStateWatch(
+        ConnectivityStateWatcherInterface* watcher) override {
+      MutexLock lock(&helper_->mu_);
+      helper_->connectivity_watchers_.erase(watcher);
+    }
+
+    void RequestConnection() override { LOG(FATAL) << "unimplemented"; }
+
+    void ResetBackoff() override { LOG(FATAL) << "unimplemented"; }
+
+    void AddDataWatcher(
+        std::unique_ptr<DataWatcherInterface> watcher) override {
+      auto* watcher_internal =
+          DownCast<InternalSubchannelDataWatcherInterface*>(watcher.get());
+      if (watcher_internal->type() == HealthProducer::Type()) {
+        AddConnectivityWatcherInternal(
+            DownCast<HealthWatcher*>(watcher_internal)->TakeWatcher());
+      } else {
+        LOG(FATAL) << "unimplemented watcher type: "
+                   << watcher_internal->type();
+      }
+    }
+
+    void CancelDataWatcher(DataWatcherInterface* watcher) override {
+      // LOG(FATAL) << "unimplemented";
+    }
+
+   private:
+    void AddConnectivityWatcherInternal(
+        std::shared_ptr<ConnectivityStateWatcherInterface> watcher) {
+      {
+        MutexLock lock(&helper_->mu_);
+        helper_->work_serializer_->Schedule(
+            [watcher]() {
+              watcher->OnConnectivityStateChange(GRPC_CHANNEL_READY,
+                                                 absl::OkStatus());
+            },
+            DEBUG_LOCATION);
+        helper_->connectivity_watchers_.insert(std::move(watcher));
+      }
+      helper_->work_serializer_->DrainQueue();
+    }
+
+    BenchmarkHelper* helper_;
+  };
+
+  class ChannelControl final
```

Comment:
Please call this something like `LbHelper`.  The `ChannelControl` prefix really just indicates what type of helper it is, not what it is.

Reply:
done